packages/components/pie-checkbox/src/index.ts

```diff
@@ -108,6 +111,23 @@ export class PieCheckbox extends FormControlMixin(RtlMixin(LitElement)) implemen
         this.handleFormAssociation();
     }
 
+    /**
+     * Called when the form that owns this component is reset.
+     * Sets the checked state to the default value.
+     */
+    public formResetCallback () : void {
+        const oldValue = this.checked;
+        this.checked = this.defaultChecked;
+
+        // If the value has changed as part of the reset, emit a change event
+        if (oldValue !== this.checked) {
+            const changeEvent = new Event('change', { bubbles: true, composed: true });
+            this.dispatchEvent(changeEvent);
+        }
```

Comment:
Not much of a difference, but may be cleaner?

```suggestion
        // If the current value is different to the default, update it and emit a change event
        if (this.checked !== this.defaultChecked) {
            this.checked = this.defaultChecked;
            const changeEvent = new Event('change', { bubbles: true, composed: true });
            this.dispatchEvent(changeEvent);
        }
```

Reply:
Nice, I like it. I actually went one step further and did an early return if `this.checked === this.defaultChecked`.